src/LatteContext/Collector/VariableCollector/AssignToTemplateVariableCollector.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Efabrica\PHPStanLatte\LatteContext\Collector\VariableCollector;
+
+use Efabrica\PHPStanLatte\LatteContext\CollectedData\CollectedVariable;
+use Efabrica\PHPStanLatte\PhpDoc\LattePhpDocResolver;
+use Efabrica\PHPStanLatte\Resolver\NameResolver\NameResolver;
+use Efabrica\PHPStanLatte\Resolver\TypeResolver\TemplateTypeResolver;
+use Efabrica\PHPStanLatte\Resolver\TypeResolver\TypeResolver;
+use PhpParser\Node\Expr\Assign;
+use PHPStan\Analyser\Scope;
+
+final class AssignToTemplateVariableCollector extends AbstractAssignVariableCollector
+{
+    private TypeResolver $typeResolver;
+
+    public function __construct(
+        LattePhpDocResolver $lattePhpDocResolver,
+        NameResolver $nameResolver,
+        TypeResolver $typeResolver,
+        TemplateTypeResolver $templateTypeResolver
+    ) {
+        parent::__construct($lattePhpDocResolver, $nameResolver, $templateTypeResolver);
+        $this->typeResolver = $typeResolver;
+    }
+
+    public function collectVariables(Assign $node, Scope $scope): array
+    {
+        $variableName = $this->getVariableName($node->var);
+        if ($variableName === null) {
+            return [];
+        }
+
+        if (!$this->isTemplateType($node->var, $scope)) {
+            return [];
+        }
+
+        $variableType = $this->typeResolver->resolveAsConstantType($node->expr, $scope);
+        if ($variableType === null) {
+            $variableType = $scope->getType($node->expr);
+        }
```

Comment:
There should be phpdoc handling but I can add it later (just create issue for me pls)

Reply:
phpdoc handling is in AbstractAssignVariableCollector
[here](https://github.com/efabrica-team/phpstan-latte/pull/205/files#diff-15d17f9cc7efa663cc2b5cc5899b6a88d80fabafbcb7a42b2161026a9618def0R46) and [here](https://github.com/efabrica-team/phpstan-latte/pull/205/files#diff-15d17f9cc7efa663cc2b5cc5899b6a88d80fabafbcb7a42b2161026a9618def0R57)

but there should be some phpdoc handling for array assign